faer-libs/faer-core/src/complex_native/c32_impl.rs

```diff
@@ -0,0 +1,1051 @@
+
+impl core::ops::Div for c32 {
+    type Output = c32;
+
+    #[allow(clippy::suspicious_arithmetic_impl)]
+    #[inline(always)]
+    fn div(self, rhs: Self) -> Self::Output {
+        self * rhs.faer_inv()
+    }
+}
+
+impl core::ops::Div<c32conj> for c32 {
+    type Output = c32;
+
+    #[allow(clippy::suspicious_arithmetic_impl)]
+    #[inline(always)]
+    fn div(self, rhs: c32conj) -> Self::Output {
+        // TODO: Can be done a bit better
+        self * rhs.canonicalize().faer_inv()
+    }
+}
+
+impl core::ops::Div<c32> for c32conj {
+    type Output = c32;
+
+    #[allow(clippy::suspicious_arithmetic_impl)]
+    #[inline(always)]
+    fn div(self, rhs: c32) -> Self::Output {
+        self * rhs.faer_inv()
+    }
+}
+
+impl core::ops::Rem<f32> for c32 {
+    type Output = c32;
+
+    #[inline(always)]
+    fn rem(self, rhs: f32) -> Self::Output {
+        Self::new(self.re % rhs, self.im % rhs)
+    }
+}
+
+impl core::ops::Rem<c32> for f32 {
+    type Output = c32;
+
+    #[inline(always)]
+    fn rem(self, rhs: c32) -> Self::Output {
+        self.rem(rhs.to_num_complex()).into()
+    }
+}
+
+impl core::ops::Rem for c32 {
+    type Output = c32;
+
+    #[inline(always)]
+    fn rem(self, rhs: Self) -> Self::Output {
+        self.to_num_complex().rem(rhs.to_num_complex()).into()
+    }
+}
+
+impl core::ops::Rem<c32conj> for c32 {
+    type Output = c32;
+
+    #[inline(always)]
+    fn rem(self, rhs: c32conj) -> Self::Output {
+        self.rem(rhs.canonicalize())
+    }
+}
+
+impl core::ops::Rem<c32> for c32conj {
+    type Output = c32;
+
+    #[inline(always)]
+    fn rem(self, rhs: c32) -> Self::Output {
+        self.canonicalize().rem(rhs)
+    }
+}
+
+impl core::ops::AddAssign<f32> for c32 {
+    #[inline(always)]
+    fn add_assign(&mut self, rhs: f32) {
+        self.re += rhs;
+    }
+}
+
+impl core::ops::AddAssign for c32 {
+    #[inline(always)]
+    fn add_assign(&mut self, rhs: c32) {
+        self.re += rhs.re;
+        self.im += rhs.im;
+    }
+}
+
+impl core::ops::AddAssign<c32conj> for c32 {
+    #[inline(always)]
+    fn add_assign(&mut self, rhs: c32conj) {
+        self.re += rhs.re;
+        self.im -= rhs.neg_im;
+    }
+}
+
+impl core::ops::SubAssign<f32> for c32 {
+    #[inline(always)]
+    fn sub_assign(&mut self, rhs: f32) {
+        self.re -= rhs;
+    }
+}
+
+impl core::ops::SubAssign for c32 {
+    #[inline(always)]
+    fn sub_assign(&mut self, rhs: c32) {
+        self.re -= rhs.re;
+        self.im -= rhs.im;
+    }
+}
+
+impl core::ops::SubAssign<c32conj> for c32 {
+    #[inline(always)]
+    fn sub_assign(&mut self, rhs: c32conj) {
+        self.re -= rhs.re;
+        self.im += rhs.neg_im;
+    }
+}
+
+impl core::ops::MulAssign<f32> for c32 {
+    #[inline(always)]
+    fn mul_assign(&mut self, rhs: f32) {
+        self.re *= rhs;
+        self.im *= rhs;
+    }
+}
+
+impl core::ops::MulAssign for c32 {
+    #[inline(always)]
+    fn mul_assign(&mut self, rhs: c32) {
+        let tmp = self.re;
```

Comment:
`*self = *self * rhs` would be less error prone and possibly faster since performing the computations in-place might make the compiler write the intermediate results to memory instead of doing everything in registers and writing it once at the end